### ISSUE TITLE
[Usage] Can't generate outputs in llava-1.6-34b

### ISSUE THREAD

**Issue by elismasilva** [haotian-liu/LLaVA#1066]
### Describe the issue

Issue: I got this error when i tried generate caption from image in liuhaotian/llava-v1.6-34b, other models with same code works normally. 

Command:
its was started with 4bit quantization: 
```python
from llava.model.builder import load_pretrained_model
tokenizer, model, image_processor, context_len = load_pretrained_model(
    model_path, model_base, model_name, load_8bit=False, load_4bit=True,device_map='cuda:0', device='cuda:0')
```
and eval with this code:

```python
    disable_torch_init()

    qs = args.query
    if model.config.mm_use_im_start_end:
        qs = (
            DEFAULT_IM_START_TOKEN
            + DEFAULT_IMAGE_TOKEN
            + DEFAULT_IM_END_TOKEN
            + "\n"
            + qs
        )
    else:
        qs = DEFAULT_IMAGE_TOKEN + "\n" + qs

    if "llama-2" in model_name.lower():
        conv_mode = "llava_llama_2"
    elif "v1" in model_name.lower():
        conv_mode = "llava_v1"
    elif "mpt" in model_name.lower():
        conv_mode = "mpt"
    else:
        conv_mode = "llava_v0"

    if args.conv_mode is not None and conv_mode != args.conv_mode:
        print(
            "[WARNING] the auto inferred conversation mode is {}, while `--conv-mode` is {}, using {}".format(
                conv_mode, args.conv_mode, args.conv_mode
            )
        )
    else:
        args.conv_mode = conv_mode

    conv = conv_templates[args.conv_mode].copy()
    conv.append_message(conv.roles[0], qs)
    conv.append_message(conv.roles[1], None)
    prompt = conv.get_prompt()

    if not isinstance(args.image_file, list):
        images = image_parser(args)
    else:
        images = load_images(args.image_file)
        
    images_tensor = process_images(images, image_processor, model.config).to(
        model.device, dtype=torch.float16
    )
    image_sizes = [image.size for image in images] 

    input_ids = tokenizer_image_token(prompt, tokenizer, IMAGE_TOKEN_INDEX, return_tensors='pt').unsqueeze(0).to(model.device)
    stop_str = conv.sep if conv.sep_style != SeparatorStyle.TWO else conv.sep2
    keywords = [stop_str]
    stopping_criteria = KeywordsStoppingCriteria(keywords, tokenizer, input_ids)
   
    with torch.inference_mode():
        output_ids = model.generate(
            input_ids,
            images=images_tensor, 
            image_sizes=image_sizes,
            do_sample=True if args.temperature > 0 else False,
            temperature=args.temperature,
            max_new_tokens=args.max_new_tokens,                                                                        
            use_cache=True,
            stopping_criteria=[stopping_criteria])
  
    outputs = tokenizer.batch_decode(output_ids, skip_special_tokens=True)[0].strip()
    
    return outputs
```

Stacktrace: 
```
  File "/home/master/miniconda3/envs/caption/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/mnt/d/LLaVA/llava/model/language_model/llava_llama.py", line 137, in generate
    return super().generate(
  File "/home/master/miniconda3/envs/caption/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/master/miniconda3/envs/caption/lib/python3.10/site-packages/transformers/generation/utils.py", line 1764, in generate
    return self.sample(
  File "/home/master/miniconda3/envs/caption/lib/python3.10/site-packages/transformers/generation/utils.py", line 2924, in sample
``    if stopping_criteria(input_ids, scores):
  File "/home/master/miniconda3/envs/caption/lib/python3.10/site-packages/transformers/generation/stopping_criteria.py", line 132, in __call__
    return any(criteria(input_ids, scores) for criteria in self)
  File "/home/master/miniconda3/envs/caption/lib/python3.10/site-packages/transformers/generation/stopping_criteria.py", line 132, in <genexpr>
    return any(criteria(input_ids, scores) for criteria in self)
  File "/mnt/d/LLaVA/llava/mm_utils.py", line 245, in __call__
    outputs.append(self.call_for_batch(output_ids[i].unsqueeze(0), scores))
  File "/mnt/d/LLaVA/llava/mm_utils.py", line 234, in call_for_batch
    if (output_ids[0, -keyword_id.shape[0]:] == keyword_id).all():
RuntimeError: The size of tensor a (2) must match the size of tensor b (3) at non-singleton dimension 0
```

**Comment by elismasilva** [comment]
As a temporary solution I added it to line 233 of mm_utils.py so that the model works, but I don't know the impact of not performing this check.:

```python
def call_for_batch(self, output_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> bool:
        offset = min(output_ids.shape[1] - self.start_len, self.max_keyword_len)
        self.keyword_ids = [keyword_id.to(output_ids.device) for keyword_id in self.keyword_ids]
-->     if 'liuhaotian/llava-v1.6-34b' not in self.tokenizer.name_or_path:
            for keyword_id in self.keyword_ids:
                if (output_ids[0, -keyword_id.shape[0]:] == keyword_id).all():
                    return True
        outputs = self.tokenizer.batch_decode(output_ids[:, -offset:], skip_special_tokens=True)[0]
        for keyword in self.keywords:
            if keyword in outputs:
                return True
        return False
```

**Comment by elismasilva** [comment]
So i found it was fixed in 1.2.2 then i will close this issue.
